### PR TITLE
update: samples to use a semantic versioning approach to select the HPCR base image

### PR DESCRIPTION
**Pull request by CarstenLeue**
Leverage the `hpcr_image` datasource from the latest https://registry.terraform.io/providers/ibm-hyper-protect/hpcr/latest/docs provider to select the HPCR image in VPC. 
This gets rid of the rather hardcoded assumption that the latest image was that returned at index `0` by the VPC REST APIs.

Also apply `terraform fmt` for consistent formatting.

### CODE REVIEW

File: terraform-hpvs/hello-world/terraform.tf [hunk 1]
```diff
@@ -116,16 +116,15 @@ resource "ibm_is_ssh_key" "hello_world_sshkey" {
...
   status     = "available"
-
 }
 
-locals {
-  # filter the available images down to the hyper protect one
-  hyper_protect_image = [for each in data.ibm_is_images.hyper_protect_images.images : each if each.os == "hyper-protect-1-0-s390x" && each.architecture == "s390x"][0]
+# locate the latest hyper protect image
+data "hpcr_image" "hyper_protect_image" {
+  images = jsonencode(data.ibm_is_images.hyper_protect_images.images)
```
[the remark this answers is not in the document]
Comment (HenryWelborn) [reply terraform-hpvs/hello-world/terraform.tf:127]: Thanks @CarstenLeue for the explanation.   `hpcr_image` is a new data source in the terraform provider.

File: terraform-hpvs/hello-world/terraform.tf [hunk 2]
```diff
@@ -138,7 +137,7 @@ resource "hpcr_contract_encrypted" "contract" {
 # construct the VSI
 resource "ibm_is_instance" "hello_world_vsi" {
   name    = format("%s-vsi", var.prefix)
-  image   = local.hyper_protect_image.id
+  image   = data.hpcr_image.hyper_protect_image.image
```
[the remark this answers is not in the document]
Comment (HenryWelborn) [reply terraform-hpvs/hello-world/terraform.tf:140]: Thanks @CarstenLeue for the explanation.   `hpcr_image` is a new data source in the terraform provider.

File: terraform-hpvs/postgresql-cluster/terraform.tf [hunk 3]
```diff
@@ -191,13 +192,13 @@ resource "ibm_is_subnet" "postgres_subnet_slave_1" {
...
 resource "ibm_is_instance" "postgres_vsi_slave_1" {
-  name    = format("%s-slave-vsi-%s", var.prefix, var.zone_slave_1)
-  image   = local.hyper_protect_image.id
-  profile = var.profile
-  keys    = ["${ibm_is_ssh_key.postgres_sshkey.id}"]
-  vpc     = ibm_is_vpc.postgres_vpc.id
-  tags    = local.tags
-  zone    = "${var.region}-${var.zone_slave_1}"
+  name      = format("%s-slave-vsi-%s", var.prefix, var.zone_slave_1)
+  image     = local.hyper_protect_image.id
```
Comment (HenryWelborn) [inline terraform-hpvs/postgresql-cluster/terraform.tf:196]: @CarstenLeue is this still `local.hyper_protect_image.id` or does it also need to switch to `data.hpcr_image.hyper_protect_image.image` ?

File: terraform-hpvs/postgresql-cluster/terraform.tf [hunk 4]
```diff
@@ -223,13 +224,13 @@ resource "ibm_is_subnet" "postgres_subnet_slave_2" {
...
 resource "ibm_is_instance" "postgres_vsi_slave_2" {
-  name    = format("%s-slave-vsi-%s", var.prefix, var.zone_slave_2)
-  image   = local.hyper_protect_image.id
-  profile = var.profile
-  keys    = ["${ibm_is_ssh_key.postgres_sshkey.id}"]
-  vpc     = ibm_is_vpc.postgres_vpc.id
-  tags    = local.tags
-  zone    = "${var.region}-${var.zone_slave_2}"
+  name      = format("%s-slave-vsi-%s", var.prefix, var.zone_slave_2)
+  image     = local.hyper_protect_image.id
```
Comment (HenryWelborn) [inline terraform-hpvs/postgresql-cluster/terraform.tf:228]: @CarstenLeue is this still `local.hyper_protect_image.id` or does it also need to switch to `data.hpcr_image.hyper_protect_image.image` ?

File: terraform-hpvs/postgresql-cluster/terraform.tf (hunk 3 above)
Comment (CarstenLeue) [reply terraform-hpvs/postgresql-cluster/terraform.tf]: good catch. I missed this. Updated in latest push

File: terraform-hpvs/postgresql-cluster/terraform.tf (hunk 4 above)
Comment (CarstenLeue) [reply terraform-hpvs/postgresql-cluster/terraform.tf]: same as above

**Review by HenryWelborn — APPROVED** [review]

**Pull request merged**